packages/rspack/src/plugins/generate-package-json-plugin.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ExecutorContext,
+  detectPackageManager,
+  serializeJson,
+  type ProjectGraph,
+} from '@nx/devkit';
+import {
+  HelperDependency,
+  createLockFile,
+  createPackageJson,
+  getHelperDependenciesFromProjectGraph,
+  getLockFileName,
+  readTsConfig,
+} from '@nx/js';
+import { type Compiler, type RspackPluginInstance } from '@rspack/core';
+import { RawSource } from 'webpack-sources';
```

Comment:
This is what the underlying `@rspack/core` package uses as there is no `rspack-sources` yet.